### PR TITLE
Introduce coordinated animation controller

### PR DESCRIPTION
**Pull request by kriskowal**
This will allow components to coordinate batch DOM reads and writes, among other things. Based on Guten Tag PR #2

### CODE REVIEW

**Comment by kriskowal** [comment]
Rebased.

File: package.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
   },
   "dependencies": {
     "animation-frame": "^0.2.5",
```
Comment (jcorbin) [inline package.json]: can we just drop this dep now?

File: hexant.js [hunk 2]
```diff
@@ -50,11 +48,12 @@ function Hexant(body, scope) {
     this.world = null;
 
     this.hash = new Hash(scope.window);
-    this.animFrame = new AnimationFrame();
+    this.animator = scope.animator.add(this);
     this.frameId = null;
```
Comment (jcorbin) [inline hexant.js]: do we need frameId anymore?

File: package.json (hunk 1 above)
Comment (kriskowal) [inline package.json]: Yes, lost that in the rebase.

File: hexant.js (hunk 2 above)
Comment (kriskowal) [inline hexant.js]: Indeed, no. Also lost that in the rebase. Moment.

**Comment by kriskowal** [comment]
Addressed

**Pull request merged**